File: 559. Maximum Depth of N-ary Tree/solution.py
"""
# Definition for a Node.
class Node:
    def __init__(self, val, children):
        self.val = val
        self.children = children
"""


class Solution:
    def maxDepth(self, root: 'Node') -> int:
        self.maxDepth = 0

        def helper(root, curDepth):
            if not root:
                return

            curDepth += 1
            self.maxDepth = max(self.maxDepth, curDepth)
            for child in root.children:
                helper(child, curDepth)
        helper(root, 0)
        return self.maxDepth

File: 559. Maximum Depth of N-ary Tree/solution.js
/**
 * // Definition for a Node.
 * function Node(val,children) {
 *    this.val = val;
 *    this.children = children;
 * };
 */
/**
 * @param {Node} root
 * @return {number}
 */
var maxDepth = function(root) {
  if (!root) return 0;
  let max = 1;
  for (let i = 0; i < root.children.length; i++) {
    max = Math.max(max, maxDepth(root.children[i]) + 1);
  }
  return max;
};
